sensor/common/collector/service_impl.go

```diff
@@ -0,0 +1,70 @@
+package collector
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/grpc/authz/idcheck"
+	"github.com/stackrox/rox/pkg/logging"
+	"google.golang.org/grpc"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type Option func(*serviceImpl)
+
+// WithAuthFuncOverride sets the AuthFuncOverride.
+func WithAuthFuncOverride(overrideFn func(context.Context, string) (context.Context, error)) Option {
+	return func(srv *serviceImpl) {
+		srv.authFuncOverride = overrideFn
+	}
+}
+
+type serviceImpl struct {
+	sensor.UnimplementedCollectorServiceServer
+
+	queue chan *sensor.ProcessSignal
+
+	authFuncOverride func(context.Context, string) (context.Context, error)
+}
+
+func newService(queue chan *sensor.ProcessSignal) Service {
+	return &serviceImpl{
+		queue: queue,
+	}
+}
```

Comment:
Anti-pattern waring - this object writes to the queue that is provided from the outside. This makes it unclear who and when should be closing this channel.

Solution: the queue should be created in the constructor of this object and returned as read-only. This object should then be responsible for closing the channel.